### ISSUE TITLE
"@typescript-eslint/prefer-optional-chain" the automatic fix is not equivalent to initial boolean expression version

### ISSUE THREAD

**Issue by TokugawaTakesi** [typescript-eslint/typescript-eslint#2224]
**Repro**

```typescript
class Repro {
  private _markupPreprocessingConfigRepresentative: null | { atLeastOneEntryPointsGroupDefinedForCurrentProjectBuildingMode: boolean } = {
      atLeastOneEntryPointsGroupDefinedForCurrentProjectBuildingMode: false
  };

  public get shouldPreprocessMarkup(): boolean {
    return this._markupPreprocessingConfigRepresentative !== null && this._markupPreprocessingConfigRepresentative.atLeastOneEntryPointsGroupDefinedForCurrentProjectBuildingMode;
  }
}
```
Currently `shouldPreprocessMarkup` always return boolean. If to allow ESLint to fix the `optional-chain issue`, in the fixed version when the `this._markupPreprocessingConfigRepresentative` is null,  `undefined` will be returned:

```typescript
public get shouldPreprocessMarkup(): boolean {
    return this._markupPreprocessingConfigRepresentative?.atLeastOneEntryPointsGroupDefinedForCurrentProjectBuildingMode;
}
```

```
TS2322: Type 'boolean | undefined' is not assignable to type 'boolean'.   Type 'undefined' is not assignable to type 'boolean'.
```

**ESLint settings**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-optional-chain": "error"
  }
}
```

**Expected Result**

TypeScript-ESLint will not cast error in above code because current `shouldPreprocessMarkup` implementation and suggested "refactoring" are not equivalent.

Alternatively, If it's impossible to refactor without conflicting with other rules, "ESLint fix" should suggest the valid refactoring.

**Actual Result**

`"@typescript-eslint/prefer-optional-chain"` error has been ocurred, but suggested refactoring ignores the `boolean` reuturn type.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.2.0` |
| `@typescript-eslint/parser`        | `3.2.0` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
Duplicate of #1893

This is a KP and limitation of the rule. This is why it doesn't have an automatic fixer, and instead has a suggestion fixer that must be manually applied by the user.
